### ISSUE TITLE
Intermittent error on continuous integration. Team City, Windows, Cypress 3.4.1

### ISSUE THREAD

**Issue by luis-m-gonzalez** [bjowes/cypress-ntlm-auth#86]
Hi,

I'd like to ask your opinion on an issue I'm having with your plug-in. I'm hesitant on calling it a bug because it may just be a problem on my particular setup.

Basically, I have this command to run the tests.
`npm run ntlm-proxy && (node_modules\.bin\cypress-ntlm run --config baseUrl=http://myurl.com & node_modules\.bin\ntlm-proxy-exit)`

Which is mostly what I copied form your guide. This works on my computer 100% of the time. However on the Team City build, it only works 60% of the time. The remaining 40%, I get this error.

```
[15:27:34][Step 5/6] Starting: D:\AGT02\temp\agentTmp\custom_script3364066283600928888.cmd
[15:27:34][Step 5/6] in directory: D:\AGT02\work\2641cc271c7d3303\CypressUiTestFolder
[15:27:35][Step 5/6] 
[15:27:35][Step 5/6] > RSPlatform.Cypress.UI.Tests@1.0.0 ntlm-proxy D:\AGT02\work\2641cc271c7d3303\CypressUiTestFolder
[15:27:35][Step 5/6] > start /min "ntlm-proxy" cmd /c node_modules\.bin\ntlm-proxy
[15:27:35][Step 5/6] 
[15:27:41][Step 5/6] ERROR: ntlm-proxy must be started before this command
[15:27:42][Step 5/6] Process exited with code 0
```

Do you have any suggestions on how to fix this? It looks like sometimes, the pipeline tries to start Cypress before the ntml-proxy is fully 'warmed up'.

**Comment by bjowes** [comment]
Hi Luis, 
Some versions ago I added a startup monitor to the plugin, such that when launching cypress it would wait up to 5 seconds for the ntlm-proxy to be available. That seemed more than enough, but your case makes me think there may be scenarios where that timeout may need to be extended. It might be the case that TeamCity isn't that optimised for launching additional background shells.

A simple way to validate this theory would be to add a timeout before you launch cypress. You could then do (on Windows):
`npm run ntlm-proxy && timeout 10 && (node_modules\.bin\cypress-ntlm run `...
This will start ntlm-proxy in a separate shell, wait 10 seconds and then launch cypress. That would give ntlm-proxy 15 seconds to launch properly. If this works for you, or at least makes the error less likely, I will add a config parameter to the plugin to set a custom timeout. The benefit is that the plugin monitors ntlm-proxy so it will only wait as long as it needs.

**Comment by luis-m-gonzalez** [comment]
Thanks, I'll try your solution. I'll have to wait a few days to make sure the problem goes away

**Comment by luis-m-gonzalez** [comment]
The timeout command doesn't work on Contiguous integration. Apparently this is a know limitation of this command, it throws the error 
`ERROR: Input redirection is not supported, exiting the process immediately.`

As a workaround I'm using `ping 127.0.0.1 -n 10 > nul` which is suggested on several stack overflow posts (other suggestions didn't work)

This is a bit of a hack, so a config parameter for the plugin might be a cleaner solution.

**Comment by bjowes** [comment]
Well that's a bummer. I'll try to get the config parameter implemented shortly. 

There is another approach that you could try - using the [node module api](https://github.com/bjowes/cypress-ntlm-auth#node-module-api) to launch cypress with the proxy. That way, the proxy is launched within the same process as cypress and the execution waits until the proxy is up. You would need to write your own launcher file (see the link) which you would then start instead of the "cypress-ntlm" script.

I'm considering rewriting the default launcher to use this approach instead of the current one, but it hasn't been around that long so I've been a bit hesitant concerning stability.

**Comment by bjowes** [comment]
Come to think of it, using the node module api **might** also solve your other issue #83 with passing config parameters to cypress.

**Comment by bjowes** [comment]
Fixed in 2.0.4 - default ntlm-proxy wait time increased to 15 seconds
